docs/content/concepts/assets/asset-checks/checking-for-data-freshness.mdx

```diff
@@ -0,0 +1,337 @@
+
+```python
+from dagster_cloud import build_anomaly_detection_freshness_checks
+
+freshness_checks = build_anomaly_detection_freshness_checks(
+  assets=[source_tables], params=None
+)
+```
+
+</TabItem>
+</TabGroup>
+
+### Step 4: Update the Definitions object
+
+The last step is to include the asset, checks, and schedule in a <PyObject object="Definitions" /> object. This enables Dagster tools to load everything we've defined:
+
+```python
+defs = Definitions(
+  assets=[source_table_freshness_checks],
+  asset_checks=[source_table_freshness_checks],
+  schedules=[source_tables_observation_schedule],
+)
+```
+
+At this point, the finished code should look like this:
+
+```python
+from dagster import (
+  build_freshness_checks_for_non_partitioned_assets,
+  multi_observable_source_asset,
+  AssetSpec,
+  Definitions,
+  MetadataValue,
+  ScheduleDefinition,
+)
+from dagster_snowflake import fetch_last_updated_timestamps, SnowflakeResource
+
+table_names = ["charges", "customers"]
+asset_specs = [AssetSpec(table_name) for table_name in table_names]
+
+
+@multi_observable_source_asset(specs=asset_specs)
+def source_tables(snowflake: SnowflakeResource):
+    with snowflake.get_connection() as conn:
+        freshness_results = fetch_last_updated_timestamps(
+            snowflake_connection=conn.cursor(), tables=table_names
+        )
+        for table_name, last_updated in freshness_results.items():
+            yield ObserveResult(
+                asset_key=table_name,
+        metadata={"dagster/last_updated_timestamp": MetadataValue.timestamp(last_updated)},
+            )
+
+
+source_tables_observation_schedule = ScheduleDefinition(
+  job=define_asset_job(
+    "source_tables_observation_job",
+    selection=AssetSelection.assets(source_tables),
+  ),
+  cron_schedule="* * * * *",
+)
+
+
+source_table_freshness_checks = build_freshness_checks_for_non_partitioned_assets(
+  assets=[source_tables],
+  maximum_lag_minutes=2 * 60,
+)
+
+
+defs = Definitions(
+  assets=[source_table_freshness_checks],
+  asset_checks=[source_table_freshness_checks],
+  schedules=[source_tables_observation_schedule],
+)
+```
+
+From here, you can [view and manually execute the checks in the Dagster UI](#freshness-checks-in-the-dagster-ui).
+
+---
+
+## Defining freshness checks for materializable assets
+
+In this section, we'll demonstrate how to implement freshness checks for materializable assets. These are assets that are materialized by a Dagster pipeline.
+
+### Step 1: Define the freshness check
+
+In this example, we used the `build_freshness_checks` function to produce an asset check that will fail if an asset’s latest materialization is more than two hours before the current time:
+
+```python
+from dagster import build_freshness_checks_for_non_partitioned_assets
+
+@asset
+def my_asset():
+  ...
+
+asset1_freshness_checks = build_freshness_checks_for_non_partitioned_assets(
+  assets=[my_asset], maximum_lag_minutes=2 * 60
+)
+```
```

Comment:
I think the OG Notion draft had a TODO for providing a better example for this section. I'll update all the code here when that's finished.